### PR TITLE
feat(table): add md-table (cdk-table with style)

### PR DESCRIPTION
**Pull request by andrewseguin**

### CODE REVIEW

File: src/lib/table/index.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+export * from './cell';
+export * from './table';
+export * from './row';
+
+@NgModule({
+  imports: [CdkDataTableModule, CommonModule],
+  exports: [MdTable, MdHeaderCell, MdCell, MdHeaderRow, MdRow],
+  declarations: [MdTable, MdHeaderCell, MdCell, MdHeaderRow, MdRow],
+})
+export class MdDataTableModule {}
```
Comment (jelbourn) [inline src/lib/table/index.ts]: Just `MdTableModule`?

File: src/lib/table/index.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import {MdCell, MdHeaderCell} from './cell';
+import {MdHeaderRow, MdRow} from './row';
+import {CommonModule} from '@angular/common';
+
+export * from './cell';
+export * from './table';
+export * from './row';
+
+@NgModule({
+  imports: [CdkDataTableModule, CommonModule],
```
Comment (jelbourn) [inline src/lib/table/index.ts]: Did you mean `MdCommonModule`?

File: src/lib/table/table.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  selector: 'md-table, mat-table',
+  templateUrl: '../core/data-table/table.html',
+  styleUrls: ['table.css'],
+  host: {
+    'class': 'mat-table',
+  },
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MdTable<T> extends CdkTable<T> { }
```
Comment (jelbourn) [inline src/lib/table/table.ts]: We should have at least one unit test for `md-table` itself

File: src/lib/table/index.ts (hunk 2 above)
Comment (andrewseguin) [inline src/lib/table/index.ts]: Using `CommonModule` for `*ngFor`

Comment (jelbourn) [inline src/lib/table/index.ts]: It should also import `MdCommonModule`

Comment (jelbourn) [inline src/lib/table/index.ts]: Also need to add the selectors to compatibility module

**Comment by andrewseguin** [comment]
All set for review

File: src/lib/table/index.ts (hunk 2 above)
Comment (andrewseguin) [inline src/lib/table/index.ts]: Whats the reason for including `MdCommonModule`?

**Comment by andrewseguin** [comment]
Updated again - added MdCommonModule and compatibility selectors

File: src/lib/table/table.ts (hunk 3 above)
Comment (jelbourn) [inline src/lib/table/table.ts]: Still needs a test

Comment (andrewseguin) [inline src/lib/table/table.ts]: Ahh right, should probably add the test to git before I commit =P There's a test now

**Comment by jelbourn** [comment]
Caretaker note: this will need a new build rule

**Pull request merged**